paymentsheet/src/main/java/com/stripe/android/common/analytics/experiment/LogLinkGlobalHoldbackExposure.kt

```diff
@@ -1,35 +1,112 @@
-            ) { "Experiments data required to log exposures" }
-            val holdbackOn = elementsSession.linkSettings?.linkGlobalHoldbackOn == true
-            eventReporter.onExperimentExposure(
-                experiment = LinkGlobalHoldback(
-                    arbId = experimentsData.arbId,
-                    group = if (holdbackOn) TREATMENT else CONTROL,
-                ),
-            )
+            CoroutineScope(workContext).launch {
+                runCatching {
+                    logExposure(elementsSession, state)
+                }.onFailure { error ->
+                    logger.error("Failed to log Global holdback exposure", error)
+                }
+            }
         }
-    }.onFailure { error ->
-        logger.error("Failed to log Global holdback exposure", error)
     }
+
+    private suspend fun logExposure(
+        elementsSession: ElementsSession,
+        state: PaymentElementLoader.State
+    ) {
+        val experimentsData = requireNotNull(
+            elementsSession.experimentsData
+        ) { "Experiments data required to log exposures" }
+        val experimentGroup: String = elementsSession.experimentsData
+            ?.experimentAssignments[LINK_GLOBAL_HOLD_BACK] ?: "control"
+
+        val customerEmail = state.getEmail()
+
+        val isReturningUser: Boolean = customerEmail?.let { isReturningUser(email = it) } == true
+
+        logger.debug(
+            """Link Global Holdback exposure: 
+                |arbId=${experimentsData.arbId},
+                |isReturningLinkConsumer=$isReturningUser,
+                |group=$experimentGroup
+            """.trimMargin()
+        )
+        eventReporter.onExperimentExposure(
+            experiment = LinkGlobalHoldback(
+                arbId = experimentsData.arbId,
+                isReturningLinkConsumer = isReturningUser,
+                group = experimentGroup,
+            ),
+        )
+    }
+
+    suspend fun isReturningUser(
+        email: String,
+    ): Boolean {
+        return linkDisabledApiRepository
+            .lookupConsumerWithoutBackendLoggingForExposure(email)
+            .map { it.exists }
+            .fold(
+                onSuccess = { it },
+                onFailure = {
+                    logger.error("Failed to check if user is returning", it)
+                    false
+                }
+            )
+    }
+
+    private suspend fun PaymentElementLoader.State.getEmail(): String? =
+        paymentMethodMetadata.linkState?.configuration?.customerInfo?.email
+            ?: config.defaultBillingDetails?.email
+            ?: customer?.let {
+                customerRepository.retrieveCustomer(
+                    CustomerRepository.CustomerInfo(
+                        id = it.id,
+                        ephemeralKeySecret = it.ephemeralKeySecret,
+                        customerSessionClientSecret = it.customerSessionClientSecret
+                    )
+                )
+            }?.email
```

Comment:
We have similar code [here](https://github.com/stripe/stripe-android/blob/master/paymentsheet/src/main/java/com/stripe/android/paymentsheet/state/PaymentElementLoader.kt#L504) to determine the actual customer email.

Can we extract that into a method and then use it in both the `PaymentElementLoader` and in here?

Reply:
created a command instead!